src/main/java/tech/zerofiltre/blog/domain/course/features/course/CourseService.java

```diff
@@ -50,14 +68,23 @@ public Course findById(long id, User viewer) throws ResourceNotFoundException, F
         Course foundCourse = courseProvider.courseOfId(id)
                 .orElseThrow(() -> new ResourceNotFoundException(THE_COURSE_WITH_ID + id + DOES_NOT_EXIST, String.valueOf(id)));
 
-
         if ((viewer == null && Status.PUBLISHED != foundCourse.getStatus())
                 || (viewer != null && !viewer.isAdmin() && isNotAuthor(viewer, foundCourse) && foundCourse.getStatus() != Status.PUBLISHED)) {
             throw new ForbiddenActionException("You are not allowed to access this course (that you do not own) as it is not yet published");
         }
         return foundCourse;
     }
 
+    public Course findByIdAndCompanyId(long id, User viewer, long companyId) throws ResourceNotFoundException, ForbiddenActionException {
```

Comment:
Si l'objectif de cette méthode est de vérifier qu'un cours serait lié à une company et le cas échéant retourner le cours, il faudrait, dans le repository qui sera utilisé par le provider, utiliser une requête jPQL avec jointure pour aller chercher celà  et retourner Optional.empty si non trouvé. Sinon retourner Optional.of(course trouvé)

Reply:
D'après la discussion de ce matin on garde la méthode pour voir si dans le temps elle est utile.